registry/storage/driver/gcs/gcs.go

```diff
@@ -294,7 +308,7 @@ func (d *driver) Reader(context context.Context, path string, offset int64) (io.
 
 			if res.StatusCode == http.StatusRequestedRangeNotSatisfiable {
 				res.Body.Close()
-				obj, err := storageStatObject(d.context(context), d.bucket, d.pathToKey(path))
+				obj, err := storageStatObject(context, d.bucket, d.pathToKey(path), d.sClient)
```

Comment:
we should make `storageStatObject` a member func rather than passing a client via struct field to a func. It's used in other places in this driver, so let's do that.